frontend/public/components/secrets/create-secret.tsx

```diff
@@ -877,22 +877,41 @@ const secretFormFactory = (secretType: SecretTypeAbstraction) => {
   }
 };
 
-const SecretLoadingWrapper = props => {
-  if (!props.obj.loaded) {
-    return <LoadingBox />;
-  }
-  const secretTypeAbstraction = toTypeAbstraction(props.obj.data);
-  const SecretFormComponent = secretFormFactory(secretTypeAbstraction);
-  const fixed = _.reduce(props.fixedKeys, (acc, k) => ({...acc, k: _.get(props.obj.data, k)}), {});
-  return <StatusBox {...props.obj}>
-    <SecretFormComponent {...props}
-      secretTypeAbstraction={secretTypeAbstraction}
-      obj={props.obj.data}
-      fixed={fixed}
-      explanation={secretFormExplanation[secretTypeAbstraction]}
-    />
-  </StatusBox>;
-};
+class SecretLoadingWrapper extends React.Component<SecretLoadingWrapperProps, SecretLoadingWrapperState> {
+  readonly state: SecretLoadingWrapperState = {
+    formComponent: null,
+    secretTypeAbstraction: SecretTypeAbstraction.generic,
+  };
+  componentDidUpdate() {
+    // Set the proper secret form component, once the secret is received by Firehose.
+    // 'formComponent' needs to be set only once, to avoid losing form state,
+    // caused by component mounting/unmounting.
+    if (!this.state.formComponent && !_.isEmpty(this.props.obj.data)) {
+      const secretTypeAbstraction = toTypeAbstraction(this.props.obj.data);
+      this.setState({
+        formComponent: secretFormFactory(secretTypeAbstraction),
+        secretTypeAbstraction,
+      });
+    }
+  }
+  render() {
+    const { obj, fixedKeys} = this.props;
+    const { secretTypeAbstraction } = this.state;
```

Comment:
nit: you could use destructuring here to avoid the assignment below on line 903

```suggestion
    const { secretTypeAbstraction, formComponent: SecretFormComponent } = this.state;
```